homeassistant/helpers/entity_platform.py

```diff
@@ -275,17 +275,24 @@ def add_entities(self, new_entities, update_before_add=False):
             device_info = entity.device_info
 
             if config_entry_id is not None and device_info is not None:
+                processed_dev_info = {
+                    'config_entry_id': config_entry_id
+                }
+                for key in (
+                        'connections',
+                        'identifiers',
+                        'manufacturer',
+                        'model',
+                        'name',
+                        'sw_version',
+                        'via_hub',
+                ):
+                    if key in device_info:
+                        processed_dev_info[key] = device_info[key]
```

Comment:
If one entity only shows its device_info with one connection, but the other entity has that connection *plus* another connection (maybe to a secondary network interface) I think the other connection would be lost.

It might be good to use a set union operation for the `connections` and `identifiers` keys.